src/facade/dragonfly_listener.cc

```diff
@@ -234,26 +234,31 @@ void Listener::PreAcceptLoop(util::ProactorBase* pb) {
   per_thread_.resize(pool()->size());
 }
 
-bool Listener::AwaitDispatches(absl::Duration timeout,
-                               const std::function<bool(util::Connection*)>& filter) {
+bool Listener::AwaitCurrentDispatches(absl::Duration timeout, util::Connection* issuer) {
   absl::Time start = absl::Now();
 
-  while (absl::Now() - start < timeout) {
-    std::atomic<bool> any_connection_dispatching = false;
-    auto cb = [&any_connection_dispatching, &filter](unsigned thread_index,
-                                                     util::Connection* conn) {
-      if (filter(conn) && static_cast<Connection*>(conn)->IsCurrentlyDispatching()) {
-        any_connection_dispatching.store(true);
+  // Fill blocking counter with ongoing dispatches
+  fb2::BlockingCounter bc{0};
+  this->TraverseConnections([bc, issuer](unsigned thread_index, util::Connection* conn) {
+    if (conn != issuer)
+      static_cast<Connection*>(conn)->AwaitCurrentDispatches(bc);
+  });
```

Comment:
An issue here is that by the time this callback gets to run on some thread, it might have already advanced to some other command, so we might be falsly waiting for one that actually doesn't have any old version of the state

But to solve this issue we'd need to track the dispatch time of each command as well as the time the global state gets updated (including on each thread if it pushes TL values)